# sample.diff
diff --git a/requests/sessions.py b/requests/sessions.py
index e34199c..8250f83 100644
--- a/requests/sessions.py
+++ b/requests/sessions.py
@@ -382,6 +382,9 @@ class Session(SessionRedirectMixin):
         method = to_native_string(method.upper())
 
         # Handle case where Content-Type is not specified in headers
+        if 'Content-Type' not in (request.headers or {}):
+            request.headers['Content-Type'] = None
+
         # Create the Request.
         req = Request(
             method=method,
diff --git a/requests/structures.py b/requests/structures.py
index 9b798b7..9953503 100644
--- a/requests/structures.py
+++ b/requests/structures.py
@@ -52,6 +52,9 @@ class CaseInsensitiveDict(MutableMapping):
     def __setitem__(self, key, value):
         # Use the lowercased key for lookups, but store the actual
         # key alongside the value.
+        if value is None:
+            del self[key]
+            return
         self._store[key.lower()] = (key, value)
 
     def __delitem__(self, key):